static/src/javascripts/projects/journalism/modules/submit-form.js

```diff
@@ -2,83 +2,128 @@
+        calloutWrapper.classList.add('success');
     });
-    enableButton(cForm);
 };
 
 const showWaiting = cForm => {
     const button = cForm.querySelector('button');
     const errorField = cForm.querySelector('.error_box');
     fastdom.write(() => {
         button.textContent = 'Sending...';
-        disableButton(button);
+        button.disabled = true;
         errorField.innerHTML = '';
     });
 };
 
-export const formatData = (elements: any) =>
+const showError = (cForm: HTMLElement, msg: string) => {
+    const errorField = cForm.querySelector('.error_box');
+    if (errorField) {
+        fastdom.write(() => {
+            errorField.innerHTML = `<p class="error">${msg}</p>`;
+        });
+    }
+    enableButton(cForm);
+};
+
+/* ---------- DATA PARSING ------------*/
+
+const readFile = (file, cForm) =>
+    new Promise(res => {
+        const reader = new FileReader();
+        reader.addEventListener(
+            'load',
+            () => {
+                const fileAsBase64 = reader.result.toString();
+                res(fileAsBase64);
+            },
+            false
+        );
+        reader.addEventListener('error', () => {
+            showError(
+                cForm,
+                'Sorry there was a problem with the file you uploaded above. Check the size and type. We only accept images, pdfs and .doc or .docx files'
+            );
+        });
+        reader.readAsDataURL(file);
+    });
+
+const getValueFromInput = (element, data) => {
+    if (isCheckbox(element)) {
+        return (data[element.name] || '').concat(`\n${element.value}`);
+    } else if (isFile(element) && element.files[0]) {
+        const cForm = element.closest('form');
+        return readFile(element.files[0], cForm);
+    }
+    return element.value;
+};
+
+export const formatData = (elements: any): Promise<any> =>
     [].reduce.call(
         elements,
-        (data, element) => {
+        async (promise, element) => {
+            const data = await promise;
             if (isNamed(element)) {
-                if (isCheckbox(element)) {
-                    data[element.name] = (data[element.name] || '').concat(
-                        `\n${element.value}`
-                    );
-                } else {
-                    data[element.name] = element.value;
-                }
+                const elementValue = await getValueFromInput(element, data);
+                data[element.name] = elementValue;
+                return data;
             }
             return data;
         },
-        {}
+        Promise.resolve({})
     );
 
-export const submitForm = (e: any) => {
+export const submitForm = async (e: any) => {
     e.preventDefault();
     const cForm = e.target;
-    const data = formatData(cForm.elements);
+    const data = await formatData(cForm.elements);
+
+    if (data['twitter-handle'].length > 0) {
```

Comment:
Is this intentional? Any post with a twitter user name is considered a robot? Even though there seems to be an element on the form to collect this? Maybe I'm missing something (likely!).